internal/storage/vector_chunk_manager.go

```diff
@@ -147,43 +150,51 @@ func (vcm *VectorChunkManager) Exist(ctx context.Context, filePath string) (bool
 }
 
 func (vcm *VectorChunkManager) readWithCache(ctx context.Context, filePath string) ([]byte, error) {
-	contents, err := vcm.vectorStorage.Read(ctx, filePath)
-	if err != nil {
-		return nil, err
-	}
-	results, err := vcm.deserializeVectorFile(filePath, contents)
-	if err != nil {
-		return nil, err
-	}
-	err = vcm.cacheStorage.Write(ctx, filePath, results)
-	if err != nil {
-		return nil, err
-	}
-	r, err := vcm.cacheStorage.Mmap(ctx, filePath)
-	if err != nil {
-		return nil, err
-	}
-	size, err := vcm.cacheStorage.Size(ctx, filePath)
-	if err != nil {
-		return nil, err
-	}
-	vcm.cacheSizeMutex.Lock()
-	vcm.cacheSize += size
-	vcm.cacheSizeMutex.Unlock()
-	if !vcm.fixSize {
-		if vcm.cacheSize < vcm.cacheLimit {
-			if vcm.cache.Len() == vcm.cache.Capacity() {
-				newSize := float32(vcm.cache.Capacity()) * 1.25
-				vcm.cache.Resize(int(newSize))
+	val, err, _ := vcm.singleflight.Do(filePath, func() (any, error) {
+		contents, err := vcm.vectorStorage.Read(ctx, filePath)
+		if err != nil {
+			return nil, err
+		}
+		results, err := vcm.deserializeVectorFile(filePath, contents)
+		if err != nil {
+			return nil, err
+		}
+		err = vcm.cacheStorage.Write(ctx, filePath, results)
+		if err != nil {
+			return nil, err
+		}
+		r, err := vcm.cacheStorage.Mmap(ctx, filePath)
+		if err != nil {
+			return nil, err
+		}
+		size, err := vcm.cacheStorage.Size(ctx, filePath)
+		if err != nil {
+			return nil, err
+		}
+		vcm.cacheSizeMutex.Lock()
```

Comment:
Will this be modified by any other execution meanwhile? Looks like the lock is not required

Reply:
Will change it to atomic value in next pr